stm/stm.ml

```diff
@@ -2703,6 +2703,8 @@ let process_transaction ~doc ?(newtip=Stateid.fresh ()) x c =
 
       | VtProofMode pm ->
         let proof_mode = Pvernac.lookup_proof_mode pm in
+        if proof_mode = None then
```

Comment:
Don't write equalities on arbitrary types, use Option.is_empty.

Reply:
After reading the code a bit more, this is probably the wrong place to do this anyways. The `VtProofMode` constructor should take a proof mode as an argument, and the check performed at the time where we go from `VernacProofMode` to `VtProofMode` in Vernac_classifier.

Reply:
Updated.  Better now?

Reply:
I think @ppedrot wants the only lookup_proof_mode to be in the classifier, changing the type of VtProofMode to pass its result to the stm.
I'm not sure he's right though, your original solution seemed fine to me.

Reply:
OK, I think I understand; I pushed alternative code in a new commit.  We'll see what @ppedrot has in mind.  And squash if that's what he wants.